gateway/src/shard/impl.rs

```diff
@@ -169,6 +170,11 @@ impl Information {
         &self.latency
     }
 
+    /// Return the session ID of the connection.
+    pub fn session_id(&self) -> Option<String> {
+        self.session_id.clone()
```

Comment:
This should return a reference via `Option::as_ref`

Reply:
Do you mean by `Option::as_deref` to return a `Option<&str>`?

Reply:
Yes, sorry